File: Algorithm/Code_Materials/python_source/charpter11_bitmap/level2/get_sum.py
# coding=utf-8

def getSum(a, b):
    while b != 0:
        sign = (a & b) << 1
        a = a ^ b
        b = sign
    return a


if __name__ == '__main__':
    print(getSum(1, -1))  # 输出: 2

File: Algorithm/Leetcode/Code_Materials/c_or_cpp/charpter11_bitmap/level2/get_sum.cpp
#include <stdio.h>
#include <stdlib.h>

int getSum(int a, int b) {
    while (b != 0) {
        int sign = (a & b) << 1;
        a = a ^ b;
        b = sign;
    }
    return a;
}


int main() {
   
 
    int test=getSum(12,13);
    
    
    return 0;
}